src/schedule/edf_schedule.c

```diff
@@ -55,6 +55,26 @@ static void schedule_edf_task_run(struct task *task, void *data)
 	}
 }
 
+static inline uint64_t edf_eval_task_deadline(struct task *task,
+					      uint64_t systick)
+{
+	struct edf_task_pdata *edf_pdata = edf_sch_get_pdata(task);
+
+	if (!edf_pdata->budget)
```

Comment:
maybe clarify with a comment in the structure definition that a budget of zero will result in a specific behavior. It's also not clear to me if this is a feature or an error condition.